### ISSUE TITLE
Undefined variable: b:syntastic_changedtick with after vim patch 7.4.2299 applied

### ISSUE THREAD

**Issue by jsivak** [scrooloose/syntastic#1862]
Getting the following error when saving a file after building vim with 7.4.2299 patch or later included:

```
Error detected while processing function <SNR>130_BufWritePostHook[4]..<SNR>130_UpdateErrors:
line   46:
E121: Undefined variable: b:syntastic_changedtick
E15: Invalid expression: [a:buf, b:syntastic_changedtick] 
```
This appears to only happen when saving .rst (ReSturcturedText files) that contain errors (or something that would cause the location window to open/be used)
I'm using the latest syntastic code from github (updated as of today).

vim builds 7.4.2298 and prior do not exhibit this issue.

**Comment by lcd047** [comment]
This looks like a messed up upgrade.  Please run `:scriptnames` and make sure you have a single installation of syntastic.  Run this in Vim:
```vim
:redir >scripts
:scriptnames
:redir END
```
Then in a terminal: 
```sh
awk '{print $2}' scripts | sort | uniq -d
```

**Comment by jsivak** [comment]
I don't get any output from the awk command you asked for.

There doesn't appear to be any duplicate scripts in the "scripts" list.

**Comment by jsivak** [comment]
Have tried fully removing and reinstalling the plugin, and still see the same behaviour.

**Comment by lcd047** [comment]
Well, the problem doesn't seem to be in syntastic.  `b:syntastic_changedtick` is referenced only [here][ref], on line 46 of `s:UpdateErrors()`, and it's set by the function [`syntastic#util#setChangedtick()`][set] called immediately above.  On the other hand, I've been using most of the recent versions of Vim, and I haven't run into this problem.  I can't reproduce it with 7.4.2305, and I can't think of any relevant change applied recently (recent patches have been minor cleanups in preparation of the release of Vim 8 anyway).  This leaves your installation. _shrug_ Perhaps look at the file `scripts` produced above and see if anything stands to attention?

[ref]: https://github.com/scrooloose/syntastic/blob/3ea6924aace91ab3b6878ec5a7b9b6e28b242f5f/plugin/syntastic.vim#L405
[set]: https://github.com/scrooloose/syntastic/blob/3ea6924aace91ab3b6878ec5a7b9b6e28b242f5f/autoload/syntastic/util.vim#L376

**Comment by jsivak** [comment]
Confirmed that "its me". I removed all plugins except syntastic and the issue no longer occurs. Will research my other plugins and see who is misbehaving. Thanks for your time.

**Comment by jsivak** [comment]
Found the the conflicting plugin: [ferret](https://github.com/wincent/ferret)

**Comment by lcd047** [comment]
I took a look at ferret.  I didn't get to the root of the problem, but I'm surprised you didn't have problems earlier running it along with syntastic.  Ferret messes with loclist windows even when you configure it to use quickfix lists.  It's pretty well written, but it certainly isn't prepared for this scenario, and neither is syntastic.  I added a note to the manual that they aren't compatible to each other (commit 72d4874).  I'll (hopefully) take a closer look at what's going on in the next few days.
